Update performance-reviews-50/README.md

natalie-hub opened the pull request:
Updated the process instructions. Removed Informal Reviews because the 90d new colleague reviews now replace a need for an informal review. Also added a placeholder for Mid-Year. Also added a note on compensation changes only during the annual Winter cycle.

<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. Learn how to update the handbook: https://handbook.mattermost.com/company/how-to-guides-for-staff/how-to-update-handbook
2. If this is your first contribution, please sign the Contributor License Agreement: http://www.mattermost.org/mattermost-contributor-agreement/
   - Once signed, you will be added to the Mattermost Approved Contributor List: https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true
   - If you included your mailing address, you may receive a Limited Edition Mattermost Mug as a thank you gift after your pull request is merged: https://forum.mattermost.org/t/limited-edition-mattermost-mugs/143
-->

#### Summary
<!--
A description of what this pull request does.
-->

natalie-hub committed:
Update performance-reviews-50/README.md

Updated the process instructions. Removed Informal Reviews because the 90d new colleague reviews now replace a need for an informal review. Also added a placeholder for Mid-Year. Also added a note on compensation changes only during the annual Winter cycle.

jasonblais commented:
@natalie-hub How would you like changes from https://github.com/mattermost/mattermost-handbook/pull/224 be incorporated with this pull request?

natalie-hub committed:
Update performance-reviews-50/README.md

Merged info from https://github.com/mattermost/mattermost-handbook/pull/224/files to this update. I will cancel PR - 224.
